src/System.Management.Automation/namespaces/RegistryProvider.cs

```diff
@@ -1825,8 +1825,19 @@ private void MoveRegistryItem(string path, string destination)
                     notePropertyName = LocalizedDefaultToken;
                 }
 
-                propertyResults.Properties.Add(new PSNoteProperty(notePropertyName, key.GetValue(valueName)));
-                valueAdded = true;
+                try
+                {
+                    propertyResults.Properties.Add(new PSNoteProperty(notePropertyName, key.GetValue(valueName)));
+                    valueAdded = true;
+                }
+                catch (InvalidCastException invalidCast)
+                {
+                    WriteError(new ErrorRecord(
+                        invalidCast,
+                        invalidCast.GetType().FullName,
+                        ErrorCategory.WriteError, 
+                        path));
```

Comment:
Could you please show screenshot with the error message?

Reply:
It is just the normal `InvalidCastException` message. I was thinking if we should use a more descriptive RESX to indicate which sub key path had issues?

Reply:
Both `SetProperty` and `NewProperty` handle cast exception the same way it seems.

Reply:
I see the code follow the file pattern so I am ok.